otlp/common.go

```diff
@@ -580,3 +583,55 @@ func shouldTrimTraceId(traceID []byte) bool {
 	}
 	return true
 }
+
+func GetSampleRate(attrs map[string]interface{}) int32 {
+	sampleRateKey := GetSampleRateKey(attrs)
+	if sampleRateKey == "" {
+		return defaultSampleRate
+	}
+
+	sampleRate := defaultSampleRate
+	sampleRateVal := attrs[sampleRateKey]
+	switch v := sampleRateVal.(type) {
+	case string:
+		if i, err := strconv.Atoi(v); err == nil {
+			if i < math.MaxInt32 {
+				sampleRate = int32(i)
+			} else {
+				sampleRate = math.MaxInt32
+			}
+		}
+	case int32:
+		sampleRate = v
+	case int:
+		if v < math.MaxInt32 {
+			sampleRate = int32(v)
+		} else {
+			sampleRate = math.MaxInt32
+		}
+	case int64:
+		if v < math.MaxInt32 {
+			sampleRate = int32(v)
+		} else {
+			sampleRate = math.MaxInt32
+		}
+	}
+	// To make sampleRate consistent between Otel and Honeycomb, we coerce all 0 values to 1 here
+	// A value of 1 means the span was not sampled
+	// For full explanation, see https://app.asana.com/0/365940753298424/1201973146987622/f
+	if sampleRate == 0 {
+		sampleRate = defaultSampleRate
+	}
+	delete(attrs, sampleRateKey) // remove attr
+	return sampleRate
+}
+
+func GetSampleRateKey(attrs map[string]interface{}) string {
```

Comment:
what do you think about expanding this to include all case variations by doing `strings.EqualFold` instead of this?

Reply:
Great idea!